File: src/tp03/ej09/Rectangle.java
package tp03.ej09;

public class Rectangle extends Figure {
    private Point corner1, corner2;

    public Rectangle(Point corner1, Point corner2) {
        this.corner1 = corner1;
        this.corner2 = corner2;
    }

    public double perimeter() {
        return 2 * base() + 2 * height();
    }

    public double area() {
        return base() * height();
    }

    public double base() {
        return Math.abs(corner1.getX() - corner2.getX());
    }

    public double height() {
        return Math.abs(corner1.getY() - corner2.getY());
    }

    @Override
    public String toString() {
       return "Rectangulo [ " + corner1 + " , " + corner2 + " ]";
    }
}

File: src/tp08/ej04/rectangle.rb
class Rectangle < Figure
  def initialize(corner1, corner2)
    @corner1 = corner1
    @corner2 = corner2
  end

  def perimeter
    2 * (@corner1.x - @corner2.x).abs + 2 * (@corner1.y - @corner2.y).abs
  end

  def area
    (@corner1.x - @corner2.x).abs * (@corner1.y - @corner2.y).abs
  end 

  def to_s
    "Rectangle [ #{@corner1} #{@corner2} ]"
  end
end
